packages/web3-utils/src/converters.ts

```diff
@@ -555,17 +558,30 @@ export const toWei = (number: Numbers, unit: EtherUnits): string => {
 	if (!denomination) {
 		throw new InvalidUnitError(unit);
 	}
+	let parsedNumber = number;
+	if (typeof parsedNumber === 'number'){
+		if (parsedNumber  < 1e-15){
+			console.warn(PrecisionLossWarning)
+		}
```

Comment:
Just wondering, why this is no need to call `parsedNumber =  BigInt(parsedNumber);` here similar to the logic for `parsedNumber  < 1e-15`?

Reply:
Large decimal numbers values become scientific notation (for example `1e-15`) and scientific notation is not accepted by BigInt. On line [573](https://github.com/web3/web3.js/pull/6908/files#diff-217301804b75cf6f71d5a808105e97fd258371e0846a38f44841668b5c852148R573) we will convert the number to a string, so there is no need to convert to a bigint